app/Database/CourseQueries.hs

```diff
@@ -168,7 +168,7 @@ buildSession lecs tuts =
 getGraphJSON :: String -> IO Response
 getGraphJSON graphName =
     runSqlite databasePath $ do
-        graphEnt :: (Maybe (Entity Graph)) <- selectFirst [GraphTitle ==. graphName] []
+        graphEnt :: (Maybe (Entity Graph)) <- selectFirst [GraphTitle ==. T.pack graphName] []
```

Comment:
Don't change this; instead, change the type of the `graphName` parameter to `T.Text`, and in the Routes.hs file, use `lookText'` (see [here](http://hackage.haskell.org/package/happstack-server-7.4.6.2/docs/Happstack-Server-RqData.html#v:lookText-39-)) instead of `look` to get the query parameter.